src/sdam/topology.ts

```diff
@@ -459,12 +461,19 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
     }
 
     const timeoutMS = this.client.options.timeoutMS;
-    const timeout = timeoutMS != null ? Timeout.expires(timeoutMS) : undefined;
+    const serverSelectionTimeoutMS = this.client.options.serverSelectionTimeoutMS;
     const readPreference = options.readPreference ?? ReadPreference.primary;
+
+    const timeoutContext = TimeoutContext.create({
+      timeoutMS,
+      serverSelectionTimeoutMS,
+      waitQueueTimeoutMS: this.client.options.waitQueueTimeoutMS
+    });
```

Comment:
```suggestion
    const timeoutContext = TimeoutContext.create(client.s.options);
```

and while we're here - can we also adjust these `client.options` calls to use `client.s.options`?